src/main/java/lotto/domain/WinningAndBonusNumbers.java

```diff
@@ -0,0 +1,22 @@
+package lotto.domain;
+
+import lotto.domain.lotto.Lotto;
+
+public class WinningAndBonusNumbers {
+    private final WinningNumbers winningNumbers;
+    private final BonusNumber bonusNumber;
+
+    private WinningAndBonusNumbers(WinningNumbers winningNumbers, BonusNumber bonusNumber) {
+        this.winningNumbers = winningNumbers;
+        this.bonusNumber = bonusNumber;
+    }
+
+    public static WinningAndBonusNumbers newWinningAndBonusNumbers(WinningNumbers winningNumbers, BonusNumber bonusNumber) {
+        return new WinningAndBonusNumbers(winningNumbers, bonusNumber);
+    }
+
+
+    public Rank rankOfLotto(Lotto lotto) {
+        return Rank.findRank(lotto.countOfMatch(winningNumbers), bonusNumber.isMatched(lotto.lottoNumbers()));
```

Comment:
값을 `Lotto`에서 꺼내서 비교하는 대신에, 메시지를 보내서 처리해보면 어떨까요?
`BonusNumber`도 결국 하나의 로또 번호이기 때문에, `LottoNumber`로 변경하면 다음과 같이 처리해볼 수도 있을 것 같아요.
```suggestion
        return Rank.findRank(lotto.countOfMatch(winningNumbers), lotto.contains(bonusNumber));
```